pkg/controller/dnszone/awsactuator.go

```diff
@@ -572,6 +572,39 @@ func (a *AWSActuator) setInsufficientCredentialsConditionToTrue(message string)
 	return accessDeniedCondsChanged
 }
 
+func (a *AWSActuator) setAPIOptInRequiredConditionToFalse() bool {
+	apiOptInConds, apiOptInCondsChanged := controllerutils.SetDNSZoneConditionWithChangeCheck(
+		a.dnsZone.Status.Conditions,
+		hivev1.APIOptInRequiredCondition,
+		corev1.ConditionFalse,
+		apiOptInNotRequiredReason,
+		"route53 apis are enabled",
+		controllerutils.UpdateConditionNever,
+	)
+	if apiOptInCondsChanged {
+		a.dnsZone.Status.Conditions = apiOptInConds
+	}
+
+	return apiOptInCondsChanged
+}
+
+func (a *AWSActuator) setAPIOptInRequiredConditionToTrue(message string) bool {
+	apiOptInConds, apiOptInCondsChanged := controllerutils.SetDNSZoneConditionWithChangeCheck(
+		a.dnsZone.Status.Conditions,
+		hivev1.APIOptInRequiredCondition,
+		corev1.ConditionTrue,
+		apiOptInRequiredReason,
+		message,
```

Comment:
Do we not need to worry about sanitizing the message to remove the UUID-type pieces from the string?

Reply:
I was more concerned about the hot-loops, but as you pointed out, this controller has been converted to rate limit on updates.